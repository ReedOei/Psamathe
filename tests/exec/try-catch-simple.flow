contract Main {}
;

try {
    revert("test")
} catch (x : ! string) {
    if x != "test" { revert("Failure!") }
}

