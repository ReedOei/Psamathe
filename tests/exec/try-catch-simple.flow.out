<generatedTop>
  <common>
    <k>
      .
    </k>
    <contracts>
      Main |-> contract Main { .Decls }
    </contracts>
    <fields>
      Main |-> .Set
    </fields>
    <constructors>
      Main |-> constructor ( .VarDefs , pass )
    </constructors>
    <transactions>
      .Map
    </transactions>
    <types>
      address |-> modified ( SetItem ( consumable )
      SetItem ( nonfungible ) , address )
      bool |-> modified ( SetItem ( consumable )
      SetItem ( nonfungible ) , bool )
      nat |-> modified ( SetItem ( consumable )
      SetItem ( fungible ) , nat )
      string |-> modified ( SetItem ( consumable )
      SetItem ( nonfungible ) , string )
    </types>
    <freshCounter>
      1
    </freshCounter>
  </common>
  <lookup>
    this |-> loc ( -1 )
  </lookup>
  <storage>
    loc ( 0 ) |-> resource ( string , "test" )
    loc ( -1 ) |-> resource ( Base , .Map )
  </storage>
  <catchStack>
    .List
  </catchStack>
</generatedTop>
